README.md

```diff
@@ -1,209 +1,404 @@
+```
+
+The returned `get` object is of type `UnaryClientCall<Echo_EchoRequest, Echo_EchoResponse>`
+and has [futures][nio-ref-elf] for the initial metadata, response, trailing
+metadata and call status. The differences between call types is detailed in
+[API Types](#api-types).
+
+Note that the call can be made synchronous by waiting on the `response` on the
+`get` object:
+
+```
+let response = try echo.get(request).response.wait()
+```
+
+We can register callbacks on the response to observe its value:
+
+```swift
+get.response.whenSuccess { response in
+  print("Received '\(response.text)' from the Echo service")
+}
+```
+
+We can also register a callback on the response to observe an error should the
+call fail:
+
+```swift
+get.response.whenFailure { error in
+  print("The Get call to the Echo service failed: \(error)")
+}
+```
+
+The call will always terminate with a status which includes a status code and
+optionally a message and trailing metadata.
+
+This is often the most useful way to determine the outcome of a call. However,
+it should be noted that __even if the call fails, the `status` future will be
+succeeded__.
+
+```swift
+get.status.whenSuccess { status in
+  if let message = status.message {
+    print("Get completed with status code \(status.code) and message '\(message)'")
+  } else {
+    print("Get completed with status code \(status.code)")
+  }
+}
+```
+
+If the program succeeds it should emit the following output (if you're running
+the example the program may terminate before the callbacks are called, to avoid
+this you can simply `wait()` on the call status):
+
+```
+Received 'Swift echo get: Hello!' from the Echo service
+Get completed with status code ok and message 'OK'
+```
+
+### Without a Generated Client
+
+It is also possible to call gRPC services without a generated client. The models
+for the requests and responses are required, however.
+
+If you are calling a service which you don't have any generated client, you can
+use `AnyServiceClient`. For example, to call "Get" on the Echo service you can
+do the following:
+
+```swift
+let connection = ... // get a ClientConnection
+let anyService = AnyServiceClient(connection: connection)
+
+let get = anyService.makeUnaryCall(
+  path: "/echo.Echo/Get",
+  request: Echo_EchoRequest.with { $0.text = "Hello!" },
+  responseType: Echo_EchoResponse.self)
+```
+
+Calls for client-, server- and bidirectional-streaming are done in a similar way
+using `makeClientStreamingCall`, `makeServerStreamingCall`, and
+`makeBidirectionalStreamingCall` respectively.
+
+These methods are also available on generated clients, allowing you to call
+methods which have been added to the service since the client was generated.
+
+### API Types
+
+Swift gRPC provides all four API styles: Unary, Server Streaming, Client
+Streaming, and Bidirectional Streaming. Calls to the generated types will return
+an object of the approriate type:
+
+- `UnaryClientCall<Request, Response>`
```

Comment:
Should we link these to the corresponding files?

Reply:
We could, but I'm not sure how much value there is, since each of them just constructs a request and a response handler before calling the superclass initializer.